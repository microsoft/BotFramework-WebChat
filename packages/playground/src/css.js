// TODO: (#3515) make into proper emotion objects
const PlaygroundStyles = `html,
body {
  font-family: "Segoe UI", "Segoe UI Web (West European)", "Segoe UI", -apple-system, BlinkMacSystemFont, Roboto, "Helvetica Neue", sans-serif;
  height: 100%;
}

body {
  background-color: #EEE;
  margin: 0;
}

#root {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: row;
}

#app-container {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: row;
}

.button-bar {
  backdrop-filter: blur(2px);
  background-color: rgba(255, 255, 255, .8);
  border-left: 1px #DDD solid;
  display: flex;
  flex-direction: column;
  min-width: 278px;
  padding: 10px;
}

fieldset {
  display: flex;
  flex-direction: column;
}

fieldset button {
  background-color: rgba(128, 128, 128, .2);
  border: 0;
  height: 25px;
  margin-bottom: 10px;
  outline: 0;
  padding: '5px 10px';
  width: 100%;
}

fieldset button:hover {
  background-color: rgba(0, 0, 0, .2);
  color: 'White'
}

.info-container {
  display: flex;
  flex-flow: row-reverse;
  justify-content: flex-end;
}

h3 {
  margin: 10px 0px 0px 0px;
}

legend {
  font-weight: bold;
  font-size: 1.5em;
}

fieldset fieldset legend {
  font-size: 1.25em;
}
.webchat {
  height: 100%;
  margin: 0;
  max-width: 500px
}

:disabled:focus,
[aria-disabled='true']:focus {
  outline-color: rgba(0, 0, 0, 0.2) !important;
}

.webchat .card__action--performed {
  background-color: #0063b1 !important;
  border-color: #0063b1 !important;
  color: White !important;
}

.webchat .card__action--performed:focus {
  outline-color: rgba(255, 255, 255, 0.6) !important;
}`;
export default PlaygroundStyles;
