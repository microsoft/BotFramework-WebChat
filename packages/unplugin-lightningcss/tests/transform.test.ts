import { resolve } from 'node:path'
import { rollupBuild, testFixtures } from '@sxzz/test-utils'
import css from 'rollup-plugin-css-only'
import { describe, expect, it } from 'vitest'
import LightningCSS from '../src/rollup'

describe('transform', async () => {
  await testFixtures(
    ['tests/fixtures/*.css'],
    async (args, id) =>
      (
        await rollupBuild(id, [
          LightningCSS({
            options: {
              minify: true,
              targets: {
                ie: 11,
              },
            },
          }),
          css(),
        ])
      ).snapshot,
    { cwd: resolve(__dirname, '..'), promise: true },
  )
})

describe('CSS Modules', () => {
  it('should transform CSS Modules', async () => {
    const entry = resolve(__dirname, './css-module-fixture/index.js')
    const { snapshot } = await rollupBuild(entry, [
      LightningCSS({
        options: {
          minify: true,
          targets: {
            ie: 11,
          },
          cssModules: {
            pattern: 'dummy_[local]',
          },
        },
      }),
      css(),
    ])
    expect(snapshot).toMatchSnapshot()
  })
})
