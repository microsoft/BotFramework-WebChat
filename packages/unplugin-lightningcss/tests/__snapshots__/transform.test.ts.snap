// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSS Modules > should transform CSS Modules 1`] = `
"// assets/asset-B_IWpIjy
.dummy_dummy_button{color:red;bakcground-color:blue}

// index.js
const button = "dummy_button";

const btn = document.createElement('div').classList.add(button);
document.body.append(btn);
"
`;

exports[`transform > tests/fixtures/basic.css 1`] = `
"// assets/asset-CuZ0KhGL
.red{color:red}

// basic.js

"
`;

exports[`transform > tests/fixtures/import.css 1`] = `
"// assets/asset-C9cPOTFW
@import "./basic.css";

// import.js

"
`;

exports[`transform > tests/fixtures/nesting.css 1`] = `
"// assets/asset-B4RRm-s2
.a .b{color:red}

// nesting.js

"
`;
